src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.logic.commands;
+
+import seedu.address.model.Model;
+
+/**
+ * A command used to undo previous undoable commands.
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Undoes the previous undoable command.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Undo successful";
+
+    @Override
+    public CommandResult execute(Model model) {
+        CommandStack commandStack = CommandStack.getInstance();
+        if (commandStack.isEmpty()) {
+            return new CommandResult("There are no commands to undo");
+        }
+        Command command = commandStack.pop();
+        boolean isUndoable = command.undo(model);
+        if (!isUndoable) {
+            return new CommandResult("The previous command is not undoable");
```

Comment:
Should we push the command back onto the commandStack? 

There could be a case where there are two consecutive commands A,B where B can only be done after A (e.g. A is add student, B is delete student). After attempting to undo B, it is popped off the stack but because B is potentially not undoable, there is no reverse effect of B. Then, we undo A which could cause some problems since the effect of 'B' has not taken place yet. E.g. in the delete student, the undo of A to delete the student could throw an unexpected error.

Reply:
Actl if we push it back onto the stack, we would not be able to undo past that command as it will always be the command being popped. Perhaps, we might need another logic to do this?